sdk/include/opentelemetry/sdk/metrics/view/predicate.h

```diff
@@ -0,0 +1,96 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#ifndef ENABLE_METRICS_PREVIEW
+#  include <vector>
+#  if (__GNUC__ == 4 && (__GNUC_MINOR__ == 8 || __GNUC_MINOR__ == 9))
+#    define HAVE_WORKING_REGEX 0
+#  else
+#    include <regex>
+#    define HAVE_WORKING_REGEX 1
+#  endif
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+#  include "opentelemetry/nostd/string_view.h"
+namespace sdk
+{
+namespace metrics
+{
+class Predicate
+{
+public:
+  virtual ~Predicate()                                                        = default;
+  virtual bool Match(opentelemetry::nostd::string_view string) const noexcept = 0;
+};
+
+class PatternPredicate : public Predicate
+{
+public:
+  PatternPredicate(opentelemetry::nostd::string_view pattern) : reg_key_{pattern.data()}
+  {
+    if (pattern == "*")
+    {
+      matchAll_ = true;
+    }
+    else
+    {
+      matchAll_ = false;
+    }
+  }
+  virtual bool Match(opentelemetry::nostd::string_view str) const noexcept override
+  {
+    if (matchAll_)
+    {
+      return true;
+    }
+    else
+    {
+#  if HAVE_WORKING_REGEX
+      return std::regex_match(str.data(), reg_key_);
+#  else
+      return false;  // not supported
```

Comment:
Seems like this should fail in some fashion.

Reply:
Yes, this will be supported eventually. For now, we can add the LOG_ERROR to log the error while returning `false`.

Reply:
Added LOG_ERROR. Let me know if you have a better way to fail. This will eventually return a default View. We don't throw exceptions in sdk. Let me know if you have a better way to fail.